importing/csv.py

```diff
@@ -17,15 +19,123 @@ def name(self) -> str:
     def suggest(self, input_path: str) -> bool:
         return input_path.endswith(".csv")
 
+    def _detect_skip_rows(self, input_path: str) -> int:
+        """Detect the number of rows to skip before CSV data begins."""
+        try:
+            with open(input_path, "r", encoding="utf8") as file:
+                lines = []
+                for i, line in enumerate(file):
+                    if i >= 50:  # Sample first 50 lines max
+                        break
+                    lines.append(line.strip())
+
+                # nothing to skip if 0 or 1 line in file
+                if len(lines) < 2:
+                    return 0
+
+                # Parse each line and analyze content
+                parsed_rows = []
+                for line in lines:
+                    if not line:  # Skip empty lines
+                        continue
+                    try:
+                        reader = csv.reader([line])
+                        row = next(reader)
+                        parsed_rows.append(row)
+                    except Exception:
+                        parsed_rows.append([line])  # Fallback for problematic lines
+
+                if len(parsed_rows) < 2:
+                    return 0
+
+                # Look for the actual CSV header (column names)
+                for i, row in enumerate(parsed_rows):
+                    if self._looks_like_csv_header(row):
+                        return i  # Skip everything before the header
+
+                # Fallback: use field count analysis
+                field_counts = [len(row) for row in parsed_rows]
+                from collections import Counter
+
+                count_frequency = Counter(field_counts)
+                most_common_count = count_frequency.most_common(1)[0][0]
+
+                # Find first row that matches the most common field count
+                for i, count in enumerate(field_counts):
+                    if count == most_common_count:
+                        return i
+
+                return 0
```

Comment:
Add validation to ensure skip_rows doesn't exceed the total file length, preventing scenarios where all data would be skipped.

```suggestion

                return 0

        except Exception:
            # If anything fails, don't skip rows
            return 0

        finally:
            # Validate result doesn't skip entire file
            if 'skip_rows' in locals() and skip_rows > 0:
                try:
                    with open(input_path, "r", encoding="utf8") as f:
                        total_lines = sum(1 for _ in f)
                    if skip_rows >= total_lines - 1:  # Need at least 1 data row
                        return 0
                except:
                    pass
```